### PR TITLE
Attachment support in Mail Providers through blob storage

### PR DESCRIPTION
**Pull request by harshitagarwal93**
Added the functionality of uploading the attachments to blob, and saving the blob reference to the Storage provider (Cosmos/Table Storage). The same attachment can be fetched during fetching the Notification Entity.

Blob structure:
```
Container: ApplicationName
│
└───Folder: NotificationType
     │
     └───Folder: NotificationId
         │   Attachment1
         │   Attachment2
         │   ...
```

### CODE REVIEW

**Review by deepika791 — APPROVED** [review]

**Review by sanjayprajap — APPROVED** [review]

File: NotificationService/NotificationService.BusinessLibrary/Business/v1/EmailManager.cs [hunk 1]
```diff
@@ -88,6 +96,7 @@ public class EmailManager : IEmailManager
             this.encryptionService = encryptionService;
             this.templateManager = templateManager;
             this.templateMerge = templateMerge;
+            this.cloudStorageClient = cloudStorageClient;
```
Comment (reddaiahnethi) [inline NotificationService/NotificationService.BusinessLibrary/Business/v1/EmailManager.cs:99]: why is this required here

Comment (harshitagarwal93) [reply NotificationService/NotificationService.BusinessLibrary/Business/v1/EmailManager.cs:99]: This can be removed. Had added it from a previous change. I'll remove it in the next iteration.
